cmds/webboot/ISO-menu.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+import (
+    "flag"
+	"log"
+    "strconv"
+    "fmt"
+    "os"
+    "os/exec"
+    "path/filepath"
+    "strings"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+
+//the entry interface record all information of a menu choise
+type Entry interface {
+	Label() string // the label will show in menu
+	Do() error  //do func will be run when the entry is chosen
+	IsDefault() bool  // IsDefault is true means this entry will be hit by default, if there is many default choise, the first on in the list will be choose. the rest will be ignored.
+}
+
+//GetChoose will present all entries into a menu with numbers, so that user can choose from them.
+func GetChoose(coord *int, introwords string, entries ...Entry) error{
```

Comment:
And actually related to that, I think it might make sense to separate the interface/menu elements from the rest of the functionality altogether. What I mean by that is instead of having `GetChoose` call a Do() method for your entry, instead have it only take a list of labels to display, and return the label that was selected. Then from here, you can mount the one that was chosen. Does that make sense? That way, all the the menu function is doing is displaying options in a menu and returning the one that was chosen. I think that would mean you don't need to have the Entry type anymore either.

Reply:
the entry interface is imitate the menu of u-root. I don't know why u-root will use this entry interface instead of a single string(I think string is much more easy too) but I guess maybe  the goal of having a Do() func is for concurrency or something?